src/HttpRequestHandler.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2021 Pei-Ru Wang, Cheryl Huang, Yuan Zhou. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * Project MyHttpServer
+ * @author Pei-Ru Wang, Cheryl Huang, Yuan Zhou
+ * @version 0.0.1
+ */
+
+#include "HttpRequestHandler.h"
+
+#include "HttpRouterHandler.h"
+
+/**
+ * HttpRequestHandler implementation
+ */
+
+HttpRequestHandler::HttpRequestHandler(boost::asio::io_context* io_context,
+                                       Server* server, const int n_count)
+    : n_count(n_count), strand(*io_context), HttpHandler(io_context) {
+  this->server = server;
+}
+
+HttpRequestHandler::~HttpRequestHandler() {
+  std::cout << n_count << ":~HandlerComplete" << std::endl;
+}
+
+void HttpRequestHandler::Start() { HandleRead(); }
+
+void HttpRequestHandler::Stop() {
+  socket.shutdown(boost::asio::ip::tcp::socket::shutdown_both);
+  socket.close();
+  server->ReleaseHandler(n_count);
```

Comment:
By the time this call returns, the `this` object has already been destructed.

Usually this is not ideal in C++. Either add comments on these call to ensure whoever reads the code understands that we cannot use `this` after this call, or delay the destruction of HttpRequestHandler.